src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(List.copyOf(shoppingCart.getProducts()));
+        orderDao.create(order);
+        shoppingCartService.clear(shoppingCart);
+        return order;
+    }
+
+    @Override
+    public List<Order> getUsersOrders(Long userId) {
+        return orderDao.getUsersOrders(userId);
+    }
+
+    @Override
+    public Order get(Long id) {
+        return orderDao.get(id).orElseThrow();
```

Comment:
```suggestion
        return orderDao.get(id).get();
```
we can simply use get() here, if order wasn't found, it will throw NoSuchElementException

Reply:
done